runtime/interpreter/value.go

```diff
@@ -2173,13 +2173,8 @@ func (v *ArrayValue) Transfer(
 	}
 
 	if res == nil {
-		res = &ArrayValue{
-			Type:             v.Type,
-			semaType:         v.semaType,
-			isResourceKinded: v.isResourceKinded,
-			array:            array,
-			isDestroyed:      v.isDestroyed,
-		}
+		res = newArrayValueFromAtreeValue(interpreter, array, v.Type)
+		res.semaType = v.semaType
```

Comment:
We would need to set the remaining fields as well:
```suggestion
		res.semaType		= v.semaType
		res.isResourceKinded	= v.isResourceKinded,
		res.isDestroyed		= v.isDestroyed,
```